File: BookWorm/UI/bookworm_desktop/lib/model/author.dart
import 'package:json_annotation/json_annotation.dart';

part 'author.g.dart';

@JsonSerializable()
class Author {
  final int id;
  final String name;
  final String biography;
  final DateTime dateOfBirth;
  final DateTime? dateOfDeath;
  final int countryId;
  final String countryName;
  final String? photoUrl;

  Author({
    this.id = 0,
    this.name = '',
    this.biography = '',
    required this.dateOfBirth,
    this.dateOfDeath,
    this.countryId = 0,
    this.countryName = '',
    this.photoUrl,
  });

  factory Author.fromJson(Map<String, dynamic> json) => _$AuthorFromJson(json);
  Map<String, dynamic> toJson() => _$AuthorToJson(this);
} 

File: BookWorm/BookWorm.Services/DataBase/Author.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookWorm.Services.DataBase
{
    public class Author
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(255)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(1000)]
        public string Biography { get; set; }
        
        [MaxLength(255)]

        public DateTime DateOfBirth { get; set; }
        
        public DateTime? DateOfDeath { get; set; }
        public int CountryId { get; set; }
        [ForeignKey(nameof(CountryId))]
        public  Country? Country { get; set; }

        [MaxLength(255)]
        public string? Website { get; set; }
        
        [MaxLength(255)]
        public string? PhotoUrl { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
        
        
        public virtual ICollection<Book> Books { get; set; } = new List<Book>();
    }
} 